rpc.go

```diff
@@ -14,6 +14,10 @@ type Message struct {
 	Params  *json.RawMessage `json:"params,omitempty"`
 	Result  *json.RawMessage `json:"result,omitempty"`
 	Error   *ErrorDetail     `json:"error,omitempty"`
+
+	// Metadata is used to store additional information about the message for
+	// processing by the client or server. It is never sent across the wire.
+	Metadata map[string]string
```

Comment:
I'm not super happy with this field, but we need a place to put metadata that isn't sent across the wire. Seemed like an okay place to put it for now